SConstruct

```diff
@@ -428,12 +432,14 @@ for variant_path in variant_paths:
         if linker:
             with gem5_scons.Configure(env) as conf:
                 if not conf.CheckLinkFlag(f'-fuse-ld={linker}'):
+                    mold_path = GetOption('mold_path')
                     # check mold support for gcc older than 12.1.0
                     if linker == 'mold' and \
                        (env['GCC'] and \
                            compareVersions(env['CXXVERSION'],
                                            "12.1.0") < 0) and \
-                       ((isdir('/usr/libexec/mold') and \
+                       ((conf.CheckLinkFlag('-B'+mold_path)) or \
```

Comment:
I think if the user specifies the mold_path and `config.CheckLinkFlag('-B'+mold_path)` fails, then we shouldn't look for alternatives, which we do via the `or`. I think if the user explicitly states path it shouldn't try alternatives. It should just fail.